Add hvac_action method to vera climate

awooganl opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->



## Proposed change
<!--
In order for a climate entity to display what the current HVAC action is, pyvera exposes the hvac_state() function.  The vera component does not implement the hvac_action property, which means that (for instance) a thermostat does not display in HA when it is heating or cooling.  This is a read-only property.

This issue was raised in #29220 
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #29220 
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

awooganl committed:
Add hvac_action method

homeassistant commented:
Hi @awooganl, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#47117).

Once you do that we will be able to review and accept this pull request.

Thanks!

bdraco committed:
Cleanup homekit and remove aid storage from hass.data (#47488)

MatsNl committed:
Improve Atag integration and bump version to 0.3.5.3 (#47778)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Store the correct context in the trace (#47785)

balloob committed:
Do not use AsyncTrackStates (#47255)

Shutgun committed:
Bump devolo_home_control_api to 0.17.0 (#47790)

bdraco committed:
Ensure homekit reset accessory service can target any entity (#47787)

emontnemery committed:
Remove unused COVER_SCHEMA from gogogate2 cover (#47170)

postlund committed:
Bump pyatv to 0.7.7 (#47798)

* Bump pyatv to 0.7.7

* Change to assume name always exist in config entry

balloob committed:
Allow filtering the logbook by context_id (#47783)

mib1185 committed:
Add ambient sensors to nut integration (#47411)

emontnemery committed:
Add temperature sensor for gogogate2 wireless door sensor (#47754)

* Add temperature sensor for gogogate2 wireless door sensor

* Chain sensor generators

cgarwood committed:
Return property_key in zwave_js get_config_parameters websocket (#47808)

iMicknl committed:
Add HomeKit support for new CO / CO2 device class (#47737)

Kane610 committed:
UniFi has changed to not report uptime in epoch form (#47492)

Kane610 committed:
Improve deCONZ init tests (#47825)

Use patch.dict rather than deep copy to change DECONZ_WEB_REQUEST

Kane610 committed:
Introduction of deCONZ websocket fixture (#47812)

klaasnicolaas committed:
Update cloud integration to 0.42.0 (#47818)

hanskroner committed:
Additional sensors for OpenWeatherMap (#47806)

cgtobi committed:
Fix Netatmo event handling (#47792)

emontnemery committed:
Fix unclean shutdown of recorder test (#47791)

dmulcahey committed:
fix exception on device removal (#47803)

balloob committed:
Bump frontend to 20210313.0 (#47844)

rajlaud committed:
Fix missing integer cast in squeezebox config flow (#47846)

allenporter committed:
Add timeouts in stream tests to prevent possible hangs (#47545)

* Add timeouts on recving packets

Add a timeout when recving packets from the worker thread in case it hangs.
Add an exit condition just in case the while loop goes on forever.

* Add a timeout to recorder thread join.

* Wait for recorder thread to be invoked in tests

Remove the while loop and instead wait for segments to be produced by the background worker thread.

* Allow worker to resume before stopping to fix timeouts

* Lower test timeout further

* Remove test_stream_ended since it is flaky

This test doesn't really add additional value on top of other tests.

raman325 committed:
Fix zwave_js preset supported feature (#47819)

tdorsey committed:
Fix spelling of automatically in roomba/lutron_caseta components (#47856)

timmo001 committed:
Update aiolyric to v1.0.6 (#47871)

postlund committed:
Add device_info to Apple TV entities (#47837)

Adminiuga committed:
Bump up ZHA dependency (#47873)

MartinHjelmare committed:
Fix zwave_js preset mode lookup (#47851)

timmo001 committed:
Update service config for lyric (#47857)

timmo001 committed:
Add HVAC action to Lyric climate platform (#47876)

frenck committed:
Verisure: Remove JSONPath, unique IDs, small cleanups (#47870)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

starkillerOG committed:
Add Xiaomi Miio subdevice lightbulb support (#46660)

* Xiaomi Miio: add subdevice lightbulb support

* fix tests

* process revieuw comments

* bump python-miio to 0.5.5

* bump python-miio to 0.5.5

* fix imports

Danielhiversen committed:
Upgrade Tibber library to 0.16.2 (#47892)

jugla committed:
Reduce number of iqair request (#47890)

danielperna84 committed:
Update pyhomematic to 0.1.72 (#47906)

balloob committed:
Bump frontend to 20210314.0

bdraco committed:
Improve error reporting in recorder purge test (#47929)

bdraco committed:
Fix homekit checking for port cleanup too many times (#47836)

* Fix homekit checking for port cleanup too many times

The loop should have terminated as soon as the port was available

* coverage

* tweak homekit shutdown wait

bdraco committed:
Add suggested area support to august (#47930)

bieniu committed:
Bump accuweather library (#47915)

bieniu committed:
Bump gios library (#47917)

unaiur committed:
Upgrade to maxcube-api-0.4.1 (#47910)

This new version implements a workaround for a hardware bug
that causes a factory reset of the full MAX! service.

See https://github.com/hackercowboy/python-maxcube-api/issues/12
for more details.

allenporter committed:
Invalidate HLS Stream on nest url refresh failure (#47869)

This will ensure that the HLS stream is re-created and fetches a new url.

bdr99 committed:
Address review comments and minor fix for Mazda integration (#47702)

* Address comments from code review

* Fix handling of missing sensor values

* Use default timeout for get_vehicles

* Fix test_update_auth_failure

pdecat committed:
Update openwrt-luci-rpc from 1.1.6 to 1.1.8 (#47848)

balloob committed:
Improve bad JSON data reporting (#47932)

* Improve bad data reporting

* Fix tests

Co-authored-by: Erik <erik@montnemery.com>

drzony committed:
MQTT Light: Use flash attribute in async_turn_off (#47919)

starkillerOG committed:
Support all Xiaomi Miio gateway switches (#46657)

* Support all gateway switches

* fix checks

* process revieuw

* Update homeassistant/components/xiaomi_miio/switch.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* generilize variable matching

* fix styling

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

starkillerOG committed:
Make Xiaomi Miio unavailable device independent (#47795)

* make unavailable independent

* fix data is None

* process review comments

Commit:
Add Hive config flow (#47300)

* Add Hive UI

* Fix tests and review updates

* Slimmed down config_flow

* Fix tests

* Updated Services.yaml with extra ui attributes

* cleanup config flow

* Update config entry

* Remove ATTR_AVAILABLE

* Fix Re-Auth  Test

* Added more tests.

* Update tests

cgtobi committed:
Add tests for Netatmo climate (#46392)

* Add tests for Netatmo climate

* Add comments and fake webhook events

* Split tests

* Split tests

* Clean up

* Fix coveragerc

* Fix requirements

* Remove freezegun dependency

* Move async_block_till_done to

* Call async_handle_webhook directly

* Use async_handle_webhook directly p2

* Exclude helper.py from

* Remove assertion of implementation details

* Use the webhook integration handler

* Extract function

bramkragten committed:
Fix target of WLED services (#47938)

colinodell committed:
Upgrade qnapstats library to 0.3.1 (#47907)

Fixes #47674

timmo001 committed:
Clean up Lyric (#47899)

alengwenus committed:
Migrate LCN configuration to ConfigEntry (Part 1) (#44090)

frenck committed:
Upgrade vsure to 1.7.3 (#47946)

dmcc committed:
Sort supported features in vlc_telnet (#46800)

MartinHjelmare committed:
Add zwave_js dev docs readme (#47621)

frenck committed:
None optional hass typing in base entity and notify (#47528)

bieniu committed:
Bump brother library (#47949)

ntilley905 committed:
Implement Wake On Lan Dummy State (#47719)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

iMicknl committed:
Update state translation strings for water_heater (#46588)

bdraco committed:
Ensure recorder purge tests can handle multiple purge cycle (#47956)

Since a purge can generate another purge task, we now wait for
three recorder queue completions by default.

ludeeus committed:
Use ClientTimeout for hassio send_command (#47957)

mib1185 committed:
improve debug logging (#47858)

RadekHvizdos committed:
Add suggested_area to MQTT discovery (#47903)

* Add suggested_area to MQTT Discovery

This adds suggested_area to MQTT discovery, so that the discovered devices could be automatically added to the proper area.

* Add abbreviation for MQTT suggested_area

* Remove extra whitespace

* Remove extra whitespace #2

* Added tests for MQTT Dicovery of suggested_area

* Fix test for MQTT suggested_area

* Better tests of MQTT suggested_area

Changes made as per feedback from @emontnemery

frenck committed:
Add config flow to Verisure (#47880)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

MartinHjelmare committed:
Add zwave_js sensor humidity device class (#47953)

frenck committed:
Move Verisure services to entity services (#47905)

frenck committed:
Add devices to Verisure integration (#47913)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

emontnemery committed:
Make it possible to list debug traces for a specific automation (#47744)

homeassistant committed:
[ci skip] Translation update

Socalix committed:
Fix xmpp notify for muc rooms (#46715)

Shutgun committed:
Add voltage device class to devolo Home Control (#47967)

frenck committed:
Add reauthentication to Verisure (#47972)

* Add reauthentication to Verisure

* Update translations

* Correct translation step key

* Address pylint warning

* = is not :

emontnemery committed:
Correct trace for choose and repeat script actions (#47973)

* Correct trace for choose and repeat script actions

* only choose-wrap the choices

* Update tests

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

emontnemery committed:
Add execute_script WS API (#47964)

* Add execute_script WS API

* Improve tests

emontnemery committed:
Add support for light color modes (#47720)

* Add support for light color modes

* Update tests

* Update comments

* Fix bugs, add tests

* Suppress lint errors

* Don't suppress brightness when state is ambiguous

* Improve reproduce_state + add tests

* Add comment

* Change COLOR_MODE_* constants, rename COLOR_MODE_DIMMER to COLOR_MODE_BRIGHTNESS

* Fix tests

* Tweaks

emontnemery committed:
Add custom JSONEncoder for automation traces (#47942)

* Add custom JSONEncoder for automation traces

* Add tests

* Update default case to include type

* Tweak

* Refactor

* Tweak

* Lint

* Update websocket_api.py

dmulcahey committed:
Guard extra call in ZHA lights (#47832)

* add flag to prevent sending an on command

* fix condition

* add constant for default transition

* make groups work with new force on flag

* reorder light entity creation

* rearrange logic

* update test

* remove failed attempt at group light flag

* fix flag

inetAnt committed:
Ignore STATE_UNKNOWN in prometheus (#47840)

Giving a value of 0 by default can lead to erroneous
data being exported. For example, if a MQTT temperature
sensor is in `STATE_UNKNOWN` (which can happen after a
HASS restart), a temperature of 0°C will be exported.
Some user might prefer no value rather than a wrong one.

frenck committed:
Add device classes to Verisure sensors (#47990)

farmio committed:
Update xknx to 0.17.3 (#47996)

emontnemery committed:
Improve JSONEncoder test coverage (#47935)

klaasnicolaas committed:
Add aliases to actions in automation blueprints (#47940)

ronaldheft committed:
Fix withings InvalidParamsException (#47975)

* Bump withings_api to 2.3.1 (fixes #47329)

* Fix NotifyAppli calls to be compatible with withings_api 2.3.1

* Fix errors with withings_api 2.2+ using pydantic

* Bump withings_api to 2.3.2

Adminiuga committed:
Bump up ZHA dependencies (#47997)

mdz committed:
Add binary_sensor entities for SmartTub reminders (#47583)

billsq committed:
Add support for Xiaomi Air Purifier Pro H (#47601)

eifinger committed:
Add missing "pin" field in step "pair" for philips_js (#47802)

chpego committed:
Upgrade youtube_dl to version 2021.03.14 (#48000)

alandtse committed:
Add location details to deprecation warning (#47155)

elupus committed:
Bump philips_js with backported fixes (#47959)

farmio committed:
KNX sensor: float no longer valid for `type` (#48005)

balloob committed:
Add run_id to automation logbook event (#47980)

abmantis committed:
Delay ZHA group updates to ensure all members are updated first (#46861)

* Delay ZHA group updates to ensure all members are updated first

After turning off a group, when the first device reports "off", the
other devices may still be "on". If HA processes the group state update
quickly enough, the group will see that some devices are on, so the
state of the group will revert back to "on", and then "off" when the
remaining devices all report "off". That would cause the UI toggle to go
back and forward quickly, and automations that trigger with "state: on"
to fire when the user turns the group off.

This PR fixes that by delaying the group state update, giving time for
all the devices to report their states first.

* Fix zha group tests

* Reorder sleeping.

* Update tests/components/zha/common.py

Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

balloob committed:
Bump frontend to 20210316.0

bdraco committed:
Remove YAML support from August (#47615)

dieselrabbit committed:
Add Pentair ScreenLogic integration (#47933)

Co-authored-by: J. Nick Koston <nick@koston.org>

homeassistant committed:
[ci skip] Translation update

spacegaier committed:
Fix historic attributes for input_datetime (#45208)

natekspencer committed:
Fix issue with setting sleep mode during DST (#48001)

blejdfist committed:
Add support for percentage based fan model in esphome (#46712)

dependabot[bot] committed:
Bump codecov/codecov-action from v1.2.2 to v1.3.1 (#48020)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.2.2 to v1.3.1.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/master/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.2.2...fcebab03f26c7530a22baa63f06b3e0515f0c7cd)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ntilley905 committed:
Add Logger Check Before Adding Another (#47954)

Commit:
Rewrite tests for Template Light (#41163)

cdce8p committed:
Update metadata license string (#46899)

schiermi committed:
Fix workday sensor to honor timezone (#47927)

mdonoughe committed:
Switch history tests to pytest (#42318)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cdce8p committed:
Update typing 01 (#48013)

hanskroner committed:
Improve OWM Precipitation sensors (#47945)

cdce8p committed:
Update typing 02 (#48014)

cdce8p committed:
Update typing 03 (#48015)

bdraco committed:
Fix rest sensor data misalignment with multiple sensors (#48043)

If there were multiple rest data sources, the index needed
to be incremented by type instead of by data source/type

glmnet committed:
Add service schema for ESPHome api services (#47426)

cdce8p committed:
Update typing 04 (#48037)

cdce8p committed:
Update typing 05 (#48038)

cdce8p committed:
Update typing 06 (#48039)

homeassistant committed:
[ci skip] Translation update

KartoffelToby committed:
Google has deprecated a comma separated list for modes changed it to array (#48029)

corneyl committed:
Upgraded aiopylgtv to v0.4.0 (#47014)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Add a service to reload config entries that can easily be called though automations (#46762)

cdce8p committed:
Update typing 07 (#48057)

cdce8p committed:
Update typing 08 (#48058)

Kane610 committed:
Use websocket fixture in deCONZ binary sensor tests (#47820)

Localize test data
Improve asserts

bdraco committed:
Reduce rest setup code (#48062)

- Switch to storing each platform config/rest data in a list

Kane610 committed:
Use websocket fixture in deCONZ climate tests (#47821)

Localize test data
Improve asserts

cdce8p committed:
Update typing 09 (#48059)

Kane610 committed:
Use websocket fixture in deCONZ cover tests (#47822)

Localize test data
Improve asserts

Kane610 committed:
Use websocket fixture in deCONZ event related tests (#47823)

Localize test data
Improve asserts

Kane610 committed:
Use websocket fixture in deCONZ fan tests (#47824)

Localize test data
Improve asserts

Kane610 committed:
Use websocket fixture in deCONZ light tests (#47826)

Localize test data
Improve asserts#

Kane610 committed:
Use websocket fixture in deCONZ lock tests (#47827)

Localize test data
Improve asserts

Kane610 committed:
Use websocket fixture in deCONZ sensor tests (#47830)

Localize test data
Improve asserts

Kane610 committed:
Use websocket fixture in deCONZ switch tests (#47831)

Localize test data
Improve asserts

Kane610 committed:
Improve deCONZ services and scenes tests (#47829)

Use patch.dict rather than deep copy to change DECONZ_WEB_REQUEST

javicalle committed:
Propagate RFLink 'send_command' event (#43588)

* propagate send_command event

* propagate send_command event

andreas-amlabs committed:
Amcrest add support for CrossLineDetection (#44582)

Co-authored-by: andreas-amlabs <andreas-amlabs@users.noreply.github.com>

cdce8p committed:
Update typing 10 (#48071)

rajlaud committed:
Allow hdmi_cec to recover from lost connection to adapter without restart (#40714)

* Only update CecDevice state when there is new data

* Replace CecDevice with CecEntity

* Support for losing and reconnecting to pycec TcpAdapter

* Register listener in async_added_to_hass

* Rename hdmi_cec watchdog

* Only update CecDevice state when there is new data

* Fix flake8 docstring error

* Fix linter error

* Bump pycec version to 0.5.0

* Bump pycec version to 0.5.1

* Fixe merge mistake

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cdce8p committed:
Update typing 11 (#48072)

bernimoses committed:
Ignore not implemented lg_soundbar source/equaliser. (#45868)

cdce8p committed:
Update typing 12 (#48073)

cdce8p committed:
Update typing 13 (#48077)

cgtobi committed:
Add tests for Netatmo oauth2 api (#46375)

* Add Netatmo tests for api

* Update tests/components/netatmo/test_api.py

* Update .coveragerc

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cdce8p committed:
Update typing 14 (#48078)

cdce8p committed:
Update typing 15 (#48079)

elyobelyob committed:
Add URL input for Prowl (#46427)

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

raman325 committed:
Bump zwave-js-server-python to 0.22.0 (#48085)

klada committed:
Add definitions for grouping media players (#41193)

* Add definitions for grouping media players

See https://github.com/home-assistant/architecture/issues/364

* Fix Google Assistant tests

* Define sync versions of async_join_players/async_unjoin

* Don't use async API in synchronous test methods

* Fix tests and make pylint happy

The method name `unjoin` is used by another component, so let's use
`unjoin_player` instead.

* Fix emulated_hue tests

The new media player entity in the `demo` component requires a tiny
adjustment in the emulated_hue tests.

* Use "target:" in service description

* Also use "name:" in service descriptions

Co-authored-by: Franck Nijhof <git@frenck.dev>

bestlibre committed:
Add images support to matrix notify (#37625)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

emontnemery committed:
Correct trace for repeat script actions (#48031)

bieniu committed:
Fix Shelly sleeping device initialization after reconfiguration (#48076)

cdce8p committed:
Update typing 16 (#48087)

cdce8p committed:
Add python-typing-update to pre-commit-config (#48088)

cdce8p committed:
Update pyupgrade to v2.10.1 (#48089)

Martidjen committed:
Add Opentherm Gateway current and setpoint precision (#47484)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

homeassistant committed:
[ci skip] Translation update

cgtobi committed:
Refactor Netatmo test (#48097)

* Refactor webhook simulate

* Update test_climate.py

farmio committed:
Type check KNX integration expose (#48055)

farmio committed:
Type check KNX integration weather, notify and scene (#48051)

farmio committed:
Type check KNX integration light (#48053)

* type check light

* review changes

farmio committed:
Type check KNX integration binary_sensor, sensor and switch (#48050)

farmio committed:
Type check KNX integration factory and schema (#48045)

these are used non-optional anyway
get them per config[] notation

bieniu committed:
Use device class voltage in NUT integration (#48096)

cgtobi committed:
Add tests for Netatmo light (#46381)

* Add tests for Netatmo light

* Improve docstring

* Register the camera data class for the light platform

* Remove freezegun dependency

* Update tests

* Update tests/components/netatmo/test_light.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Deduplicate webhook test data

* Mock pytest to verify it is called

* Don't test internals

* Rename

* Assert light still on with erroneous event data

Co-authored-by: Erik Montnemery <erik@montnemery.com>

cgtobi committed:
Add tests for Netatmo camera (#46380)

* Add test for Netatmo camera

* Improve docstrings

* Remove light tests

* Remove freezegun from tests

* Update camera tests

* Remove freezegun dependency

* Update tests/components/netatmo/test_camera.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Update tests/components/netatmo/test_camera.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Deduplication of the fake webhook payload

* Mock pyatmo instead of checking the logs

* Clean up

* Further deduplication

* Assert function arguments

* Rename mocked functions

* Update .coveragerc

Co-authored-by: Erik Montnemery <erik@montnemery.com>

andreasbrett committed:
Add "timestamp" attribute to seventeentrack (#47960)

* bump py17track to 3.0.1

* Make aiohttp ClientSession optional

as introduced in py17track v3.0.0 (https://github.com/bachya/py17track/releases/tag/3.0.0)

* Update manifest.json

* add new attribute timestamp

introduced in 3.1.0

* Update requirements.txt

* Update requirements_all.txt

* Update requirements.txt

* Update requirements_test_all.txt

* Update sensor.py

* Update sensor.py

* Update manifest.json

* provide timezone configuration

user config to pre-define timezone of package status timestamps

* Update requirements_all.txt

* Update requirements_test_all.txt

* linting

* use hass.config.time_zone

* Update sensor.py

* Update test_sensor.py

* Update test_sensor.py

* black

* Update manifest.json

* adjust changes to session param

* added test against latest dev branch

* make isort happy

* make black happy

* make flake8 happy

* make black happy

* bump to 3.2.1

* 3.2.1

* Update typing 15

miccico committed:
Fix Homematic transition function on light devices with multiple channels (#45725)

* Update light.py

Fix Transition function on devices with multiple channels

* Update light.py

fix Flake8 Warning W293 blank line contains whitespace

emontnemery committed:
Improve websocket debug log

frenck committed:
Upgrade RPi.GPIO to 0.7.1a4 (#48106)

Kane610 committed:
Improve test coverage of deCONZ config flow (#48091)

frenck committed:
Add flake8 comprehensions checks to pre-commit & CI (#48111)

MartinHjelmare committed:
Add missing oauth2 error abort reason (#48112)

bdraco committed:
Have pylint warn when user visible log messages do not start with capital letter or end with a period (#48064)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

javicalle committed:
Wait for switch startup in generic_thermostat (#45253)

* Better status control on restore

* Better status control on restore

* fix code coverage

* Rollback hvac_mode initialization

I think I have better understood the handling of the `hvac_mode`.
I change the approach. Now the thermostat doesn't initialize until the switch is available.

* fix pyupgrade

* fix black

* Delete test_turn_on_while_restarting

HVAC mode should not be modified by the switch.
IMHO, this test does not make sense because if the switch is turned on the thermostat is not turning on (and not changing HVAC_MODE)

* Re add turn off if HVAC is off

If HVAC_MODE is off thermostat will not control heater switch. This can be because `initial_hvac_mode`, because state defaults to or because old_state.
IMHO it is preferable to be excessively cautious.

* Update climate.py

* Change warning message

* Fix black

* Fix black

Commit:
Improve uvc test camera (#41438)

* Improve uvc test camera

* Clean setup full config

* Clean setup partial config

* Set more camera defaults

* Clean setup partial config v31x

* Clean setup incomplete config

* Clean setup nvr errors during indexing

* Clean setup nvr errors during initialization

* Clean properties

* Fix motion recording mode properties

* Clean stream

* Clean login

* Clean login v31x

* Clean login tries both addres and caches

* Clean login fails both properly

* Remove not needed test

* Clean camera image error

* Test snapshot login retry

* Clean up

* Test enable and disable motion detection

* Times must be UTC

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Kane610 committed:
Improve sensor coverage by verifying daylight sensor attributes (#48090)

pavoni committed:
Make Vera should_poll static rather than dynamic (#47854)

* Make should_poll static.

* Address review comments.

* Fix black error.

farmio committed:
Type check KNX integration fan (#48056)

farmio committed:
Type check KNX integration cover (#48046)

frenck committed:
Remove defunct test from percentage util (#48125)

homeassistant committed:
[ci skip] Translation update

chilicheech committed:
Set zwave_js climate precision to tenths for F (#48133)

frenck committed:
Rewrite of not a == b occurances (#48132)

decompil3d committed:
Update pyvesync to 1.3.1 (#48128)

cgtobi committed:
Add tests for Netatmo webhook handler (#46396)

* Add tests for Netatmo webhook handler

* Add async prefix

* Remove freezegun dependency

* Clean up

frenck committed:
Yoda assertion style removed is (#48142)

Kane610 committed:
Improve test coverage of deconz_device (#48141)

emontnemery committed:
Warn on undefined variables in templates (#48140)

* Warn on undefined variables in templates

* Add test

* fix tests

* fix tests

frenck committed:
Fix a collection of tests with missing asserts (#48127)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

dmulcahey committed:
only block coord removal if it is active (#48147)

spacegaier committed:
Use domain const in config_flow (#48168)

homeassistant committed:
[ci skip] Translation update

OttoWinter committed:
ESPHome trigger reconnect immediately when mDNS record received (#48129)

javicalle committed:
Update RFLink tests (#48149)

frenck committed:
Small code styling tweaks for HomeKit (#48163)

javicalle committed:
Make Rflink handle set_level command for dimmable devices (#46499)

* Added handle_event for set_level command in dimmable devices

* refactor common code for dimmable devices

* Force tests

Silly change to force tests execution

* fix super()

* add rflink dim utils

emontnemery committed:
Mark base components' state_attribute @final, rename others to extra_state_attributes (#48161)

* Mark base state_attributes @final, rename others to extra_state_attributes

* Fix calendar, update tests

dieselrabbit committed:
ScreenLogic cleanups (#48136)

* ScreenLogic cleanup.
Bump screenlogicpy to 0.2.0.
Move heating functions from water_heater to climate platform.
Address notes from original PR.

* Fix temperature attribute

* Addressing notes.
Bump screenlogicpy to 0.2.1.
Made device_types constants.
Made (known) equipment flags constants.
Used dict.get() in places where None is the default.
Return fast with good _last_preset.

* Update homeassistant/components/screenlogic/climate.py

Let base entity handle state property.

Co-authored-by: J. Nick Koston <nick@koston.org>

* Patch integration setup functions.

* Exception, ATTR_TEMPERATURE notes

Co-authored-by: J. Nick Koston <nick@koston.org>

eltociear committed:
Fix typo in homekit strings.json (#48176)

Co-authored-by: J. Nick Koston <nick@koston.org>

xonestonex committed:
Preset support for MOES thermostat valves (#48178)

heythisisnate committed:
Handle switch state updates from Konnected device (#48167)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Index config entries by id (#48199)

bdraco committed:
Update homekit to improve representation of activity based remotes (#47261)

bdraco committed:
Make sure include_ignore=False always works with _async_current_entries (#48196)

If the step was anything other than SOURCE_USER,
include_ignore=False would not be honored

emlove committed:
Fix Kulersky and Zerproc config unloading (#47572)

emontnemery committed:
Improve condition trace tests (#48152)

emontnemery committed:
Test that homeassistant stop and restart do not block WS (#48081)

jjlawren committed:
Trigger Plex GDM scans regularly (#48041)

bdraco committed:
Exclude homekit accessories created by the homekit integration from homekit_controller (#48201)

bdraco committed:
Increase config entries test coverage (#48203)

bdraco committed:
Convert august to be push instead of poll (#47544)

bdraco committed:
Ensure homekit yaml config works when there is an ignored config entry (#48175)

balloob committed:
Populate trigger variable when manually triggering automation (#48202)

* Populate trigger variable when manually triggering automation

* Update tests/components/automation/test_init.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

bdraco committed:
Bump yalexs to 1.1.5 for august (#48205)

Turns on auto-reconnect support

pavoni committed:
Remove vera should_poll (#48209)

swilson committed:
Update aqualogic library to v2.6 (#48119)

emontnemery committed:
Migrate integrations a-c to extend SensorEntity (#48210)

emontnemery committed:
Migrate integrations d-e to extend SensorEntity (#48211)

frenck committed:
Bump colorlog to 4.8.0 (#48218)

frenck committed:
Upgrade pre-commit to 2.11.1 (#48219)

unaiur committed:
Fix maxcube temperature for thermostat auto mode (#48047)

maxcube-api dependency now supports using None as the
target temperature: in that case, it uses the scheduled
temperature in auto mode and current temperature in all
other modes.

We will use that feature when changing hvac_mode to auto and
selecting PRESET_NONE.

emontnemery committed:
Improve script tracing (#48100)

* Improve script tracing

* Fix test

frenck committed:
Upgrade pyupgrade to v2.11.0 (#48220)

starkillerOG committed:
Add identification for YAML imports (#48162)

MatsNl committed:
Add jobstate parser to Onvif integration (#46589)

spacegaier committed:
Fix condition extra fields for climate and humidifier (#48184)

dermotduffy committed:
Add an option to hide selected Hyperion effects (#45689)

emontnemery committed:
Refactor tracing: Move trace support to its own integration (#48224)

emontnemery committed:
Migrate integrations f-h to extend SensorEntity (#48212)

emontnemery committed:
Migrate integrations n-q to extend SensorEntity (#48214)

emontnemery committed:
Migrate integrations t-v to extend SensorEntity (#48216)

emontnemery committed:
Migrate integrations w-z to extend SensorEntity (#48217)

emontnemery committed:
Migrate integrations r-s to extend SensorEntity (#48215)

emontnemery committed:
Migrate integrations i-m to extend SensorEntity (#48213)

plomosits committed:
Improve Docker and Kubernetes support for KNX (#48065)

Co-authored-by: Matthias Alphart <farmio@alphart.net>

emontnemery committed:
Move SensorEntity last in the inheritance tree (#48230)

balloob committed:
Clean up AsusWRT (#48012)

homeassistant committed:
[ci skip] Translation update

uvjustin committed:
Remove login details before logging stream source (#45398)

* Remove login details before logging stream source

* Convert to str before re

* Use compiled RE

* Add tests and filter log message in worker

* Update import

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* isort

Co-authored-by: Erik Montnemery <erik@montnemery.com>

emontnemery commented:
@awooganl the change looks good, however it's difficult to review without knowledge of the underlying link or API. Could you share a link to that?

Also, please update the tests in components/vera/test_climate.py to cover the new code.

dewet22 committed:
Bump nanoleaf to 0.1.0, add unique IDs (#48135)

* bump pynanoleaf and expose model/serial as unique_id

* addressed PR feedback

fronzbot committed:
Add Blink config migration (#46671)

awooganl commented:
Hi @emontnemery , thank you for your feedback.  The relevant information is contained in my reply to the original issue here: https://github.com/home-assistant/core/issues/29220#issuecomment-562348506 but basically, my change is to add a call to PyVera to retrieve the hvac state so it can show in HA if the climate device is Heating or Cooling or Idle.  You can find the implementation of PyVera here: https://github.com/pavoni/pyvera/blob/941823e1a13c8ea2d4dd398287939bf0c6f970f6/pyvera/__init__.py#L1303

I will update the test.

keis committed:
Change nanoleaf name to configured name instead of hostname (#46407)

* nanoleaf: Key config by device id

Rather than host which is not stable

* nanoleaf: Use pretty name instead of hostname

awooganl commented:
> Also, please update the tests in components/vera/test_climate.py to cover the new code.

I've just had a look, but there are no tests at all for the Vera component.  Are you asking me to create a new file with tests for a bug fix?  I'm not the maintainer of the component...

emontnemery commented:
There is a file named `tests/components/vera/test_climate.py`, isn't that for this integration?

awooganl commented:
> There is a file named `tests/components/vera/test_climate.py`, isn't that for this integration?

My sincere apologies, I am new to this type of development and could not see it in the components/vera directory.  Will add now!

emontnemery commented:
No worries!

To run the tests locally, follow the guide here: https://developers.home-assistant.io/docs/development_testing

Note that `tox` takes forever, you can just do `pytest tests/components/vera/test_climate.py` and let the full `tox` suite be done in CI.

awooganl committed:
Resolve conflict on climate.py and include tests for current HVAC action

awooganl committed:
Resolve conflict on climate.py and include tests for current HVAC action

awooganl commented:
Not sure what I can do about the failing check as they do not appear to be related at all to my changes?

frenck commented:
@awooganl There are unrelated commits in this PR. You'll need to git rebase it onto the latest dev branch with only your own commits.

awooganl commented:
I'm sorry @frenck , I'm really struggling with git on this and as a consequence I've no idea how I got unrelated commits into it.  I'm now not sure how further to proceed with this.

emontnemery commented:
@awooganl this is a nice guide for how to keep your branch up to date: 
 https://gist.github.com/Chaser324/ce0505fbed06b947d962 and avoid problems when merging or rebasing.

In the current state, the simplest way forward is for you to create a new local branch off latest dev, also named `issue_29220`, cherry pick https://github.com/home-assistant/core/pull/47117/commits/3a6bff9237a2cc48047e20b0b41b5e87b0a9bd7a to it and then force push it.

You can also just close this PR and create a new one.
